source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,325 @@
+
+.. code-block:: none
+   :copyable: false
+
+   Error: couldn't connect to server 127.0.0.1:27017
+
+The following sections describe methods that may help resolve the issue.
+
+.. _node-troubleshooting-connection-string-port:
+
+Check Connection String
+-----------------------
+
+Verify that the hostname and port number in the connection string are both
+accurate. In the sample error message, the hostname is ``127.0.0.1`` and the
+port is ``27017``. The default port value for a MongoDB instance is
+``27017``, but you can configure MongoDB to communicate on another port.
+
+.. _node-troubleshooting-connection-firewall:
+
+Configure Firewall
+------------------
+
+Assuming that your MongoDB deployment uses the default port, verify that port
+``27017`` is open in your firewall. If your deployment uses a different port,
+verify the correct port is open in your firewall.
+
+.. warning::
+
+   Do not open a port in your firewall unless you are sure that it's the port
+   used by your MongoDB instance.
+
+ECONNREFUSED Error
+~~~~~~~~~~~~~~~~~~
+
+If the connection is refused when the driver attempts to connect to the MongoDB
+instance, it generates this error message:
+
+.. code-block:: none
+   :copyable: false
+
+   MongoServerSelectionError: connect ECONNREFUSED ::1:27017
+
+The following sections describe methods that may help resolve the issue.
+
+Start mongod in IPV6 Mode
+-------------------------
+
+Beginning with version 17, Node.js defaults to using ``IPV6``. You can start
+your ``mongod`` in ``IPV6`` mode with
+:manual:`the --ipv6 flag </reference/program/mongod/#std-option-mongod.--ipv6>`,
+or setting the ``net.ipv6`` option to ``true``
+:manual:`in your configuration file </reference/configuration-options/#mongodb-setting-net.ipv6>`.
+
+.. code-block:: sh
+   :caption: Starting mongod with the --ipv6 flag
+
+   mongod --ipv6
+
+.. code-block:: yaml
+   :caption: Setting the net.ipv6 option in the configuration file
+
+   net:
+     ipv6: true
+
+Use IPV4 with Your Mongo Client
+-------------------------------
+
+Beginning with version 17, Node.js defaults to using ``IPV6``. You can
+explicitly use ``IPV4`` with your client by specifying ``family: 4`` as an
+`option to your MongoClient <{+api+}/interfaces/MongoClientOptions.html#family>`__.
+
+.. code-block:: js
+
+   const client = new MongoClient(uri, {
+     family: 4,
+   });
+
+ECONNRESET Error
+~~~~~~~~~~~~~~~~
+
+If the connection is reset when the driver calls ``client.connect()``, it
+generates this error message:
+
+.. code-block:: none
+   :copyable: false
+
+   MongoServerSelectionError: connect ECONNRESET ::1:27017
+
+The following section describes a method that may help resolve the issue.
+
+Increase Number of File Descriptors
+-----------------------------------
+
+In most operating systems, each connection is associated with a `file
+descriptor
+<https://www.computerhope.com/jargon/f/file-descriptor.htm>`_. There is
```

Comment:
I thought the link was approved since this was a copy from the old FAQ (https://www.mongodb.com/docs/drivers/node/current/faq/#what-can-i-do-if-i-m-getting--econnreset--when-calling--client.connect----) and moved over (as opposed to completely new authored content). I will remove the link when I reword the section.